src/kouta_indeksoija_service/rest/organisaatio.clj

```diff
@@ -1,77 +1,88 @@
-     :organisaatiot
-     (map #(:oid %))
-     vec)))
+  (->> (get-json-and-handle-errors (resolve-url :organisaatio-service.v2.hae.tyyppi)
+                                   {:aktiiviset true :suunnitellut false :lakkautetut false :organsaatiotyyppi "Oppilaitos"})
+       :organisaatiot
+       (map #(:oid %))
+       vec))
 
 (defn get-tyyppi-hierarkia
   [oid]
-  (with-error-logging
-   (get->json-body (resolve-url :organisaatio-service.v2.hierarkia.tyyppi)
-                   {:aktiiviset true :suunnitellut true :lakkautetut true :oid oid})))
+  (get-json-and-handle-errors (resolve-url :organisaatio-service.v2.hierarkia.tyyppi)
+                              {:aktiiviset true :suunnitellut true :lakkautetut true :oid oid}))
 
 (defn get-hierarkia-v4
   [oid & {:as params}]
-  (get->json-body (resolve-url :organisaatio-service.v4.hierarkia.hae)
-                  (merge {:aktiiviset true :suunnitellut true :lakkautetut false :oid oid :skipParents false} params)))
+  (get-json-and-handle-errors (resolve-url :organisaatio-service.v4.hierarkia.hae)
+                              (merge {:aktiiviset true :suunnitellut true :lakkautetut false :oid oid :skipParents false} params)))
 
 (defn find-last-changes
   [last-modified]
-  (with-error-logging
-    (let [date-string (long->date-time-string last-modified)]
-      (let [res (->> (get->json-body (resolve-url :organisaatio-service.v2.muutetut.oid) {:lastModifiedSince date-string})
-                     (:oids)
-                     (remove clojure.string/blank?)
-                     (vec))]
-        (when (seq res)
-         (log/info "Found " (count res) " changes since " date-string " from organisaatiopalvelu"))
-        res))))
+  (let [date-string (long->date-time-string last-modified)]
+    (let [res (->> (get-json-and-handle-errors (resolve-url :organisaatio-service.v2.muutetut.oid) {:lastModifiedSince date-string})
+                   (:oids)
+                   (remove clojure.string/blank?)
+                   (vec))]
+      (when (seq res)
+        (log/info "Found " (count res) " changes since " date-string " from organisaatiopalvelu"))
+      res)))
 
 (defn find-by-oids
   [oids]
-  (if (empty? oids) []
-    (with-error-logging
-     (log/debug (str "Calling organisaatio service find-by-oids") )
-     (let [url (resolve-url :organisaatio-service.v4.find-by-oids)
-           body (str "[\"" (string/join "\", \"" oids) "\"]")]
-       (:body (post url {:body body :content-type :json :as :json}))))))
+  (if (empty? oids)
+    []
+    (let [url (resolve-url :organisaatio-service.v4.find-by-oids)
+          body (str "[\"" (string/join "\", \"" oids) "\"]")]
+      (log/debug "POST => " url)
+      (let [response (http/post url {:body body :content-type :json :as :json :throw-exceptions false})
+            status   (:status response)
+            body     (:body response)]
+        (cond
```

Comment:
Voisikohan tämänkin virheenkäsittelyn yleistää POST-kutsuille vrt. get-json-and-handle-errors, se on aika kiva.